### PR TITLE
feat(server): implement postgres queue driver

### PR DESCRIPTION
**Pull request by schoren**
This PR implements a postgres queue driver

## Changes

-

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/executor/queue_driver_in_memory.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"log"
+)
+
+type loggerFn func(string, ...any)
+
+func newLoggerFn(name string) loggerFn {
+	return func(format string, params ...any) {
+		log.Printf("[%s] %s", name, fmt.Sprintf(format, params...))
+	}
+}
```
Comment (danielbdias) [inline server/executor/queue_driver_in_memory.go:14]: This is useful even outside of this package. Perhaps we could move it to a specific infra package in the future?

File: server/executor/queue_driver_postgres.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		Channel: ch.name,
+		Job:     job,
+	})
+
+	if err != nil {
+		ch.log("error marshalling pgJob: %s", err.Error())
+		return
+	}
+
+	ctx, cancelCtx := context.WithTimeout(context.Background(), 500*time.Millisecond)
```
Comment (danielbdias) [inline server/executor/queue_driver_postgres.go:165]: Can we add this `500ms` timeout as a constant? Something like `enqueueTimeout`?

**Review by danielbdias — APPROVED** [review]

File: server/executor/queue_driver_in_memory.go (hunk 1 above)
Comment (schoren) [reply server/executor/queue_driver_in_memory.go:14]: yes, I found the pattern very useful, but if we want to improve our logging I would make it a bit more sophisticated

File: server/executor/queue_driver_postgres.go (hunk 2 above)
Comment (schoren) [reply server/executor/queue_driver_postgres.go]: done!

**Review by xoscar — APPROVED** [review]

**Pull request merged**